src/lib/replication/log_store/home_raft_log_store.cpp

```diff
@@ -0,0 +1,268 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+
+#include "home_raft_log_store.h"
+#include "storage_engine_buffer.h"
+#include <sisl/fds/utils.hpp>
+
+using namespace homestore;
+
+SISL_LOGGING_DECL(replication)
+
+#define REPL_STORE_LOG(level, msg, ...)                                                                                \
+    LOG##level##MOD_FMT(replication, ([&](fmt::memory_buffer& buf, const char* msgcb, auto&&... args) -> bool {        \
+                            fmt::vformat_to(fmt::appender{buf}, fmt::string_view{"[{}:{}] "},                          \
+                                            fmt::make_format_args(file_name(__FILE__), __LINE__));                     \
+                            fmt::vformat_to(fmt::appender{buf}, fmt::string_view{"[{}={}] "},                          \
+                                            fmt::make_format_args("replstore", m_logstore_id));                        \
+                            fmt::vformat_to(fmt::appender{buf}, fmt::string_view{msgcb},                               \
+                                            fmt::make_format_args(std::forward< decltype(args) >(args)...));           \
+                            return true;                                                                               \
+                        }),                                                                                            \
+                        msg, ##__VA_ARGS__);
+
+namespace homestore {
+static constexpr store_lsn_t to_store_lsn(uint64_t raft_lsn) { return s_cast< store_lsn_t >(raft_lsn) - 1; }
```

Comment:
Homestore logstore id is int64_t because it uses streamtracker which is int64_t. It is 0 based index and first invalid id is -1 (and not 0). 0 is a valid lsn, whereas raft 0 is an invalid lsn.